components/emoji_picker/emoji_picker.test.jsx

```diff
@@ -20,4 +39,119 @@ describe('components/emoji_picker/EmojiPicker', () => {
             expect(filterEmojiSearchInput(testCase.input)).toEqual(testCase.output);
         });
     });
+
+    const actions = {
+        getCustomEmojis: jest.fn(),
+        incrementEmojiPickerPage: jest.fn(),
+        searchCustomEmojis: jest.fn(),
+    };
+
+    const customEmojis = new Map();
+    customEmojis.set('smiley', {name: 'smiley'});
+
+    const emojiMap = new Set();
+    emojiMap.add({customEmojis});
+    emojiMap.add({customEmojisArray: ['smiley']});
+
+    const smiley = new Map();
+    smiley.set('smiley', {fileName: '1f603', category: 'people', aliases: ['smiley']});
+
+    emojiMap.add(smiley);
+
+    const baseProps = {
+        customEmojiPage: 1,
+        customEmojisEnabled: true,
+        listHeight: 245,
+        onEmojiClick: jest.fn(),
+        onEmojiClose: jest.fn(),
+        emojiMap: new EmojiMap([]),
+        recentEmojis: [],
+        actions,
+    };
+
+    test('init snapshot', () => {
+        const wrapper = shallow(
+            <EmojiPicker {...baseProps}/>
+        );
+
+        wrapper.instance().emojiPickerContainer = {
+            offsetHeight: 200,
+        };
+
+        expect(wrapper).toMatchSnapshot();
+
+        // Nine categories as there is no recent caterogry
+        expect(wrapper.find(EmojiPickerCategory).length).toBe(9);
+        expect(wrapper.find(EmojiPickerCategory).find({selected: true}).length).toBe(1);
+
+        // People EmojiPickerCategory should have prop selected true
+        expect(wrapper.find(EmojiPickerCategory).find({selected: true}).find({category: 'people'}).length).toBe(1);
+
+        expect(wrapper.find(EmojiPickerItem).length).toBe(291);
```

Comment:
I think this is can be broken rather easily by adding or removing default emoji's as in when updating the emoji set, is there another way to make it resistant?

Reply:
Ya, Actually we are not interested in the actual numbers here so made changes to remove these